### PR TITLE
fix(tsconfig): enforce `verbatimModuleSyntax` and fix files

### PR DESCRIPTION
**Pull request by fernandomg**
# Description:

By enabling `verbatimModuleSyntax` we will be forced to specify when we're importing a `type`.

Using the `verbatimModuleSyntax` flag on a clean install triggered an error upon `pnpm wagmi-generate` script call.

So to keep the usefulness of the `tsc` checks I moved away the check from the `pre-commit` husky hook and moved it to a `pre-push` husky hook.

# Steps:

```bash
# update husky config
$ pnpm I # or pnpm prepare

# modify a file with a type import (remove the `type` keyword)

# commit the file
$ git commit -a -m 'chore: a broken file'

# push and pray for the hook to work and see something like the following
$ git push
Running TS checks...
src/components/sharedComponents/BigNumberInput.tsx:2:3 - error TS1484: 'RefObject' is a type and must be imported using a type-only import when 'verbatimModuleSyntax' is enabled.

2   RefObject,
    ~~~~~~~~~


Found 1 error in src/components/sharedComponents/BigNumberInput.tsx:2

TS checks failed. Push aborted.
error: failed to push some refs to 'github.com:bootnodedev/dAppBooster.git'
```

Edit: THANKS @gabitoesmiapodo for the husky help 💡

## Type of change:

- [x] New feature
- [ ] Bug fix
- [x] Breaking change
- [x] Enhancement
- [x] Refactoring
- [x] Chore

# How Has This Been Tested?

- [x] Manual testing
- [ ] Automated tests
- [ ] Other (explain)

# Remember to check that:

- Your code follows the style guidelines of this project
- You have performed a self-review of your code
- You have commented your code in hard-to-understand areas
- You have made corresponding changes to the documentation
- Your changes generate no new warnings

# Screenshots

(Add screenshots or videos to help test this pull request)

### CODE REVIEW

**Comment by fernandomg** [comment]
@nicosampler please re-check this, as it's completely different from what you approved

File: githooks/pre-push [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
```
Comment (gabitoesmiapodo) [inline githooks/pre-push:1]: Just to keep consistency, this should be into `.husky/pre-push`

**Review by gabitoesmiapodo — CHANGES_REQUESTED** [review]
Close, but no cigar.

**Review by gabitoesmiapodo — APPROVED** [review]
Made a few extra changes to streamline the workflow a bit, but I think we're all set.

**Pull request merged**